packages/node_modules/nav-frontend-veileder/src/veileder.tsx

```diff
@@ -42,90 +35,80 @@ interface FrameStyle {
     height?: number;
 };
 
-class Veileder extends React.Component<VeilederProps> { // eslint-disable-line react/prefer-stateless-function
-    public static defaultProps: Partial<VeilederProps> = {
-        type: 'normal'
+export interface NavVeilederSnakkebobleProps {
+    sier: React.ReactNode | React.ReactChild | React.ReactChildren;
+    offset: number;
+};
+
+const NavVeilederSnakkeboble: React.StatelessComponent<NavVeilederSnakkebobleProps> = props => {
+    const { sier, offset } = props;
+
+    const scaleMultiplier = Math.max(0.5, offset / 150);
+
+    const bubbleStyles = {
+        bottom: offset + ((20 * scaleMultiplier) / 2),
+        marginRight: offset / 2
     };
 
+    const arrowStyles = {
+        bottom: Math.ceil(-20 - (((20 * scaleMultiplier) / 2) - 10)),
+        right: ((20 - (20 * scaleMultiplier)) * -1) + (20 * scaleMultiplier),
+        transform: `scale(${scaleMultiplier})`
+    };
+
+    return (
+        <div className="nav-veileder__snakkeboble" style={bubbleStyles}>
```

Comment:
Det bør tas en avgjørelse på hvordan vi skal forholde oss til `style` bruken i nav-frontend.

I starten ble det sagt at less-modulene var nav-frontend, slik at nav-frontend ikke skulle være så låst til en spesifik teknologi (react). Allerede i dag finnes det unntak (modal f.eks), mens spørsmålet er om vi skal fortsette ned den samme veien. Eller prøve å holde oss til original-tanken. 

Grunnen til at `style` blir problematisk er at hvis noen ønsker å bruke komponenten på en nav-side som er servert fra enonic så er ikke det rett-frem. 

Føler kanskje at flere bør ytre litt meninger om dette før vi merger denne. 

@erlendev @frodehansen2 ?

Reply:
Ah, ok 👍 Er kanskje ikke så ille om vi overlater styling av scale til bruker? Kan evt. skrive om til at komponenten kommer med noen predefinerte størrelser (ala. 'S', 'M', 'L', 'XL' osv).

Reply:
Å knytte stylinga av komponentene tett opp imot React er generelt noe vi bør unngå så langt det er mulig, det er tanken om teknologiuavhengighet ift. js som vel er grunnen til at vi har oppdelingen i egne React- og -style-moduler. Jeg er ikke så veldig langt inne i implementasjonen her - hva er grunnen til at man trenger en `offset`-prop @Lillebo? Er det pga. forskjellige størrelser på illustrasjonene som sendes inn?

Reply:
Så ikke svaret ditt @Lillebo før jeg hadde svart via review'en min. Vi bør heller har predefinerte størrelser på disse panelene, enn at vi skal prøve å støtte alle størrelser. Da unngår vi samtidig `offset`-prop og React-spesifikk styling.

Reply:
@erlendev Hm, jau. `offset` er den egendefinerte bredden på `NavVeileder`-komponenten som blir sendt inn til `NavVeilederSnakkeboble` for å posisjonere denne korrekt relativt til størrelsen til `NavVeileder`. Er kun disse komponentene som per nå bruker React-styling. `VeilederPanel` opererer ikke med noe slikt :)

Reply:
@nutgaard @erlendev Trenger vi å diskutere dette ytterligere, eller skal jeg bare begynne å skrive om til å ha predefinerte størrelser uten noe React-styling?

Reply:
Jeg tror ikke dette trenger ytterligere diskusjon @Lillebo @nutgaard, men nøyaktig hvilke predefinerte størrelser vi skal ha er kanskje en sak for @sergiohaisch? Foreslår at man kun lager størrelser for dem det er behov for idag (og evt. ber folk endre om de forskjellige variantene er alt for mange), slik at vi ikke lager support for flere varianter enn vi faktisk trenger :)